scene/3d/ray_cast_3d.cpp

```diff
@@ -185,7 +193,17 @@ void RayCast3D::_notification(int p_what) {
 			if (prev_collision_state != collided && get_tree()->is_debugging_collisions_hint()) {
 				if (debug_material.is_valid()) {
 					Ref<StandardMaterial3D> line_material = static_cast<Ref<StandardMaterial3D>>(debug_material);
-					line_material->set_albedo(collided ? Color(1.0, 0, 0) : Color(1.0, 0.8, 0.6));
+
+					Color collide_color;
+					// Mix the base color with an highlight color to show collision
+					if ((debug_shape_color.get_h() < 0.055 || debug_shape_color.get_h() > 0.945) && debug_shape_color.get_s() > 0.5 && debug_shape_color.get_v() > 0.5) {
+						// If base color is already quite reddish, hightlight collision with a greener color
+						collide_color = Color(0.0, 1.0, 0.0, debug_shape_color.a);
+					} else {
+						// Else, hightlight collision with a more red color
+						collide_color = Color(1.0, 0, 0, debug_shape_color.a);
+					}
+					line_material->set_albedo(collided ? collide_color : debug_shape_color);
```

Comment:
Based on my comment with `_update_debug_shape_material`, here you could just replace the whole block with:
```
_update_debug_material(true);
```